src/pages/global-authorization/forms/ResourceGlobalForm.js

```diff
@@ -0,0 +1,146 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { DataSets } from 'src/resources/DataSets'
+
+// ** Custom Imports
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { AccessControlRepository } from 'src/repositories/AccessControlRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+
+export default function ResourceGlobalForm ({
+  labels,
+  maxAccess,
+  resourceName,
+  resourceId,
+  moduleId
+})
+{
+
+    const [isLoading, setIsLoading] = useState(false)
+
+    const [initialValues, setInitialData] = useState({
+      resourceId: resourceId,
+      resourceName: resourceName,
+      accessLevel: '',
+      moduleId: moduleId,
+      accessLevelName: ''
+   })
+
+    const { getRequest, postRequest } = useContext(RequestsContext)
+
+    const invalidate = useInvalidate({
+        endpointId: SystemRepository.ModuleClassRES.qry
+      })
+
+    const formik = useFormik({
+        initialValues,
+        enableReinitialize: true,
+        validateOnChange: true,
+        validationSchema: yup.object({
+          accessLevel: yup.string().required(' ')
+        }),
+        onSubmit: async obj => {
+          
+          const response = await postRequest({
+            extension: AccessControlRepository.AuthorizationResourceGlobal.set,
+            record: JSON.stringify(obj)
+          })
+
+          toast.success('Record Edited Successfully')
+          invalidate()
+        }
+      })
+
+      useEffect(() => {
+        ;(async function () {
+          try {
+
+            if (resourceId) {
+              setIsLoading(true)
+
+              const res = await getRequest({
```

Comment:
useResourceQuery instead of this?

Reply:
this is get not query, i have never realize using it for such a case..